flyteplugins/go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -141,9 +142,29 @@ func serviceAccountName(metadata pluginsCore.TaskExecutionMetadata) string {
 	return name
 }
 
-func createSparkPodSpec(taskCtx pluginsCore.TaskExecutionContext, podSpec *v1.PodSpec, container *v1.Container) *sparkOp.SparkPodSpec {
-	annotations := utils.UnionMaps(config.GetK8sPluginConfig().DefaultAnnotations, utils.CopyMap(taskCtx.TaskExecutionMetadata().GetAnnotations()))
-	labels := utils.UnionMaps(config.GetK8sPluginConfig().DefaultLabels, utils.CopyMap(taskCtx.TaskExecutionMetadata().GetLabels()))
+func createSparkPodSpec(
+	taskCtx pluginsCore.TaskExecutionContext,
+	podSpec *v1.PodSpec,
+	container *v1.Container,
+	k8sPod *core.K8SPod,
+) *sparkOp.SparkPodSpec {
+
+	annotations := pluginsUtils.UnionMaps(
+		config.GetK8sPluginConfig().DefaultAnnotations,
+		pluginsUtils.CopyMap(taskCtx.TaskExecutionMetadata().GetAnnotations()),
+	)
+	labels := pluginsUtils.UnionMaps(
+		config.GetK8sPluginConfig().DefaultLabels,
+		pluginsUtils.CopyMap(taskCtx.TaskExecutionMetadata().GetLabels()),
+	)
+	if k8sPod != nil && k8sPod.GetMetadata() != nil {
+		if k8sPod.Metadata.Annotations != nil {
+			annotations = pluginsUtils.UnionMaps(annotations, k8sPod.GetMetadata().GetAnnotations())
+		}
+		if k8sPod.Metadata.Labels != nil {
+			labels = pluginsUtils.UnionMaps(labels, k8sPod.GetMetadata().GetLabels())
+		}
```

Comment:
nit: there is a nil check in `GetMetadata()`, so we don't need to do it here
```suggestion
	if k8sPod.GetMetadata().GetAnnotations() != nil {
		annotations = pluginsUtils.UnionMaps(annotations, k8sPod.GetMetadata().GetAnnotations())
	}
	if k8sPod.GetMetadata().GetLabels() != nil {
		labels = pluginsUtils.UnionMaps(labels, k8sPod.GetMetadata().GetLabels())
		}
```

Reply:
Fixed. Thanks!